python/pyarrow/tests/test_fs.py

```diff
@@ -23,39 +25,276 @@
 
 import pytest
 
+import pyarrow as pa
 from pyarrow import ArrowIOError
+from pyarrow.tests.test_io import gzip_compress, gzip_decompress
 from pyarrow.fs import (FileType, Selector, FileSystem, LocalFileSystem,
                         SubTreeFileSystem)
-from pyarrow.tests.test_io import gzip_compress, gzip_decompress
+
+
+class FileSystemWrapper:
+
+    # Whether the filesystem may "implicitly" create intermediate directories
```

Comment:
The testing approach here looks overkill to me. All filesystem operations are extensively tested on the C++ side, we don't need to re-test them on the Python side. Just verify that the wrappers work.

Reply:
@pitrou You mean that we should not use a parallel implementation (like the Minio client for S3 and pathlib for Local), and just use the bindings in the test case?

Reply:
Yes. At least for S3. You can have a couple sanity checks for Local if you want.